lib/compiler/doc/src/compile.xml

```diff
@@ -772,29 +772,44 @@ module.beam: module.erl \
           </item>
         </taglist>
 
-	<p>Another class of warnings is generated by the compiler
-	  during optimization and code generation. They warn about
-	  patterns that will never match (such as <c>a=b</c>), guards
-	  that always evaluate to false, and expressions that
-	  always fail (such as <c>atom+42</c>).</p>
-	  <p>Those warnings cannot be disabled (except by
-	  disabling all warnings).</p>
+	<p>Another kind of warnings is <em>opportunistic
+	warnings</em>. They are generated when the compiler happens to
+	notice potential issues during optimization and code
+	generation.</p>
 
-	  <note>
-	    <p>The compiler does not warn for expressions that it
+	<note>
+	  <p>The compiler does not warn for expressions that it
           does not attempt to optimize. For example, the compiler tries
           to evaluate <c>1/0</c>, detects that it will cause an
 	  exception, and emits a warning. However,
 	  the compiler is silent about the similar expression,
```

Comment:
```suggestion
	  the compiler is silent about similar expression
```

Reply:
Looking closer at the entire paragraph (written by me a long time ago), I found it very clumsy and decided I needed to rewrite it:

The compiler does not warn for expressions that it does not attempt to optimize. For example, the compiler will emit a warning for `1/0` but not for `X/0`, because `1/0` is a constant expression that the compiler will attempt to evaluate.